### ISSUE TITLE
Rituals and Shaman Practices

### ISSUE THREAD

**Issue by Maldaris** [Maldaris/grugcraft#1]
**Goal**
Create a set of ancillary progression that augments the traditional TFC experience, and adds new challenges and customization for players and player groups, with PVP in mind.

**Features**
- Define a set of rituals discovered through specific unique tasks that players must complete to unlock the ability to create an associated item conveying a
buff or other tangible benefit

**Comment by DerFlammenwerfer** [comment]
**Ritual Categories**
Religions throughout history have been widely varied in their practices, and this addon will attempt to capture this in a fairly simple and intuitive manner. For this purpose, there will be a number of broad categories which the ritual system shall encompass, defining the general requirements of the altar, as well as the sorts of reagents and practices of functional rituals. Polytheistic religions may encompass multiple categories

- Celestial: Some of the earliest religions known to man worshipped the sun, the moon, and the stars. Rituals of this kind draw their power from time itself, requiring fewer or even no reagents in some cases, but generally being restricted to certain times of the day and/or gaining bonuses from certain celestial events. Altars to celestial bodies universally require a completely open ceiling above a pedestal. For example, Solar rituals can be performed at any time during the day, but are stronger and cheaper within 30 minutes of midday during clear weather, while Lunar rituals may be performed at any time during the night, but are most powerful during the Full Moon phase.

- Natural: The second most common religions of ancient times venerated the natural world, with certain plants and animals important to a tribe's survival become regarded as sacred, or indeed supernatural forces believed to be associated with nature, such as Faerie Folk, being entreated to by the likes of Druids and Wicca. Shrines to Nature typically involve circles of trees or saplings, flowers, and in some cases even mushrooms or cacti that occur naturally in the biome the altar is constructed in, as well as a pool of water to act as a conduit for mana to items placed in the pool.

- Elemental: Another common theme of religion venerates specific, powerful forces observed in the natural world and the circumstances that create them. Most commonly, these forces can be narrowed down to four 'primordial' elements; Earth, Air, Water, and Fire. Earth shrines are commonly erected in low elevations, such as Caves, and built of stone. Air shrines are typically found in high elevations, such as mountain tops, and feature open walls. Water shrines are common near rivers and oceans, and Fire shrines are found in arid, hot environments such as Deserts and Savannahs.

- Aethereal: Religions that claim to draw their power from 'righteous' and 'good' spirits and gods fall under this category. These religions tend to refer to themselves as using 'white' or 'holy' magic, and are often ironically decadent, making use of Quartz, stained glass, and precious metals/minerals in their construction.

- Spiritual: Religions that draw their power from empyrean powers, but do not care for the concepts of 'good' or 'evil' fall under this category. Some examples of this include venerating the spirits of one's ancestors, the neutral concepts of life and/or death, and gods that encompass specific facets of human civilization that are neither inherently good or evil, such as War, Love, Festival, Artifice, and Pleasure. The shrines and altars of these religions are the most widely varied of all, but most of them incorporate materials from the End as conduits.

- Infernal: Religions that draw their power from diabolical or 'dark' powers, or which encourage its followers to engage in activities that are widely regarded by society to be impious and/or self-indulgent fall under this category. In many cases, the practitioners of these religions keep their shrines small enough to be hidden, as they may not be welcome within their own society or able to openly practice their faith without fear of retribution, and these altars will often use materials from the Nether (excepting Nether Quartz) and from corpses.

**Comment by Maldaris** [comment]
Problem with Infernal: iirc TFC disables the nether.

**Comment by Maldaris** [comment]
**Ritual Methods**
There's a significant breadth of potential ritual practices, with different potentials for scaling. To encourage cooperation and growth of nation-states, Rituals should scale primarily on the number of players involved in the practice, and unlock new types/levels at significant digits of player interactions.

- **Single Caster:** Provides single scope, short lived/"one-off" buffs or penalties to targets. Typically imbued in items, or with a combination of somatic and material components. Should be significant enough so that spamming is _mostly_ off the table, especially in PVP, but PVE/Resource Processing spells could allow for batch/repeated use.

- **Multiple Casters (2-5):** Provides multiple scope, finite duration buffs or penalties to targets. Centered around a single altar/athame, with a ritual "leader" and several supporting "acolytes". Somatic components are supplied by the acolytes, with the leader offering the ritual materials on the altar.

- **Devotion Casters (10-50):** Provides region-wide, permanent changes to the biome, including (but not limited to) converting chunks to nether (infernal), replenishing a biome of its biological resources (natural), generation of specific ore veins (elemental), conversion of a biome to a different type (celestial), domain-wide combat buffs for a State (Aetherial), and Soulbound enchantments for single items. (Spiritual). Should require multiple altars, with multiple leaders & acolytes at each altar, with geographical positioning playing an important role in the ritual. Lines between the altars should trace out the symbol Domain of the spell being cast. Material components should be interspersed between all altars, with one "key component" at the altar where the effect should be centered (allowing for manipulating the position of the Domain symbol in cases where the symbol cannot be centered).

**State Religion**
States integration wherein States can *permanently* adopt a given ritual category, and after performing a Devotion ritual, receive bonuses/pay less for rituals in the given domain. The symbol center needs to be in the capital municipality of the State. Similarly, devotion rituals of other categories are either less effective or cost more materials depending on the effect.

**Comment by DerFlammenwerfer** [comment]
Devotion Casters should have a number of 24h buffs they can perform as daily prayers, at least three different ones per domain.

**Comment by Maldaris** [comment]
> Devotion Casters should have a number of 24h buffs they can perform as daily prayers, at least three different ones per domain.

If you mean State Religions, yeah that's a good idea. Devotion Casters is supposed be a class of ritual.

**Comment by DerFlammenwerfer** [comment]
> > Devotion Casters should have a number of 24h buffs they can perform as daily prayers, at least three different ones per domain.
> 
> If you mean State Religions, yeah that's a good idea. Devotion Casters is supposed be a class of ritual.

Yes. Ideally, an established State Religion may perform Congregation rituals in its Holy City.

A Devotion Ritual for the Nature Domain would have two effects. It would apply bonemeal to all crops and saplings in the region (twice if the State Religion is in the Nature domain) and, if a fully-grown animal is sacrificed, the empowered ritual will cause up to 10 of that animal in the region to enter breeding mode, even if they have previously already been bred.
